apps/prairielearn/src/lib/config.ts

```diff
@@ -216,6 +216,11 @@ const ConfigSchema = z.object({
    */
   workersCount: z.number().nullable().default(null),
   workersPerCpu: z.number().default(1),
+  /**
+   * If set to true, the code caller pool worker count will default to '0',
+   * and workers will be created lazily.
+   */
+  workersAreLazy: z.boolean().default(false),
```

Comment:
Rather than make this a bona fide config option, why not make it an argument to `init()` in `code-caller/index.ts` and then change the call in `helperServer.before()`?

Reply:
I can do that if preferred -- is there some reasoning against a config flag? There are already a couple of worker configuration flags.

Reply:
Config options are mostly introduced when something should be configurable in production environments. We'd never want this to be configurable there; if workers can't be created, we should error out immediately instead of booting up and health-checking as healthy.

More generally, I think it's best to minimize the number of actual config options so that self-hosters don't have to guess at what's needed (and so that we don't have to explain all these options to them or anyone else!).

Reply:
Changed in 8d538e3259672ebdf95a39c7e5955ff5af949a63.